src/repo.rs

```diff
@@ -356,6 +419,32 @@ impl<T: NetworkAdapter + 'static> Repo<T> {
                     panic!("Document closed with outstanding handles.");
                 }
             }
+            RepoEvent::ConnectNetworkAdapter(repo_id, adapter) => {
+                self.network_adapters
+                    .entry(repo_id.clone())
+                    .and_modify(|_| {
+                        // TODO: close the existing stream/sink?
```

Comment:
We could drain all messages on an existing stream and close it(I think), and poll close the sink.